vxyowsup/whatsapp.py

```diff
@@ -40,21 +40,25 @@ class WhatsAppTransport(Transport):
     CONFIG_CLASS = WhatsAppTransportConfig
     transport_type = 'whatsapp'
 
+    @defer.inlineCallbacks
     def setup_transport(self):
         config = self.get_static_config()
         log.info('Transport starting with: %s' % (config,))
         CREDENTIALS = (config.phone, config.password)
 
         stack_client = self.stack_client = StackClient(CREDENTIALS, self)
-
         self.client_d = deferToThread(stack_client.client_start)
         self.client_d.addErrback(self.catch_exit)
         self.client_d.addErrback(self.print_error)
 
+        self.redis = yield TxRedisManager.from_config(config.redis_manager)
+        self.redis = self.redis.sub_manager(self.transport_name)
```

Comment:
We should probably set up redis first and tear it down last because our other code will be using it.